scipy/ndimage/tests/test_filters.py

```diff
@@ -692,91 +693,75 @@ def test_gauss_memory_overlap(self):
+                             [(ndimage.gaussian_filter, 0, 1.0),
+                              (ndimage.uniform_filter, 1, 3.0),
+                              (ndimage.minimum_filter, 1, 3.0),
+                              (ndimage.maximum_filter, 1, 3.0)])
     @pytest.mark.parametrize(
         'axes',
         tuple(itertools.combinations(range(-3, 3), 1))
         + tuple(itertools.combinations(range(-3, 3), 2))
         + ((0, 1, 2),))
-    def test_filter_axes(self, axes, function_name):
+    def test_filter_axes(self, filter_func, size0, size, axes):
         array = numpy.arange(6 * 8 * 12, dtype=numpy.float64).reshape(6, 8, 12)
-        filter_func = getattr(ndimage, function_name)
-        if function_name == 'gaussian_filter':
-            sigma = 1.0
-            filter_args = (array, sigma)
-        elif function_name in ('uniform_filter', 'minimum_filter',
-                               'maximum_filter'):
-            size = 3
-            filter_args = (array, size)
 
         axes = tuple(ax % array.ndim for ax in axes)
         if len(tuple(set(axes))) != len(axes):
             # parametrized cases with duplicate axes raise an error
-            with pytest.raises(ValueError):
-                filter_func(*filter_args, axes=axes)
+            with pytest.raises(ValueError, match="axes must be unique"):
+                filter_func(array, size, axes=axes)
             return
-        output = filter_func(*filter_args, axes=axes)
-
-        if function_name == 'gaussian_filter':
-            # result should be equivalent to sigma=0.0 on unfiltered axes
-            all_sigmas = (sigma if ax in axes else 0.0
-                          for ax in range(array.ndim))
-            filter_args = (all_sigmas,)
-        elif function_name in ('uniform_filter', 'minimum_filter',
-                               'maximum_filter'):
-            # result should be equivalent to size=1 on any unfiltered axes
-            all_sizes = tuple(size if ax in axes else 1
-                              for ax in range(array.ndim))
-            filter_args = (all_sizes,)
-        expected = filter_func(array, *filter_args)
+        output = filter_func(array, size, axes=axes)
+
+        # result should be equivalent to sigma=0.0/size=1 on unfiltered axes
+        all_sizes = (size if ax in axes else size0 for ax in range(array.ndim))
+        expected = filter_func(array, all_sizes)
         assert_allclose(output, expected)
 
-    @pytest.mark.parametrize(
-        'axes', tuple(itertools.combinations(range(-3, 3), 2))
-    )
-    def test_gauss_axes_kwargs(self, axes):
+    kwargs_gauss = dict(radius=[4, 2, 3], order=[0, 1, 2],
+                        mode=['reflect', 'nearest', 'constant'])
+    kwargs_other = dict(origin=(-1, 0, 1),
+                        mode=['reflect', 'nearest', 'constant'])
+    @pytest.mark.parametrize("filter_func, size0, size, kwargs",
+                             [(ndimage.gaussian_filter, 0, 2, kwargs_gauss),
```

Comment:
For consistency with above.
```suggestion
                             [(ndimage.gaussian_filter, 0, 1.0, kwargs_gauss),
```